src/coreclr/src/dlls/mscoree/coreclr/CMakeLists.txt

```diff
@@ -43,11 +43,11 @@ else(CLR_CMAKE_HOST_WIN32)
         set(START_WHOLE_ARCHIVE -Wl,--whole-archive)
         set(END_WHOLE_ARCHIVE -Wl,--no-whole-archive)
 
-        if(CLR_CMAKE_TARGET_SUNOS)
+        if(CLR_CMAKE_TARGET_SUNOS AND LD_SOLARIS)
```

Comment:
Since you are touching this and we have the same code at many places, would you mind extracting this as a cmake function and using it everywhere?

Reply:
Extracted it in `set_exports_linker_option` function.